backend/walkthrough

```diff
@@ -0,0 +1,222 @@
+
```

Comment:
Ok this is the explanation I needed before!
Thanks!
However this does not belong in this commit as this is a documentation this should be added in a DOC commit.

Or even better, in the project wiki :)